### PR TITLE
Generated fixtures won't use parent_id when generated with parent:references

### LINKED ISSUE #18301: Generated fixtures shouldn't include parent_id when generated with parent:references
I created two models:

```yaml
#test/fixtures/comments.yml
pablo:
  commenter: Pablo
  body: MyText
  post_id: 

benja:
  commenter: Benja
  body: MyText
  post_id: 
```

```yaml
#test/fixtures/posts.yml

about:
  title: MyString
  body: MyText

second:
  title: MyString
  body: MyText
```

After using the scaffold, I read the documentation about associations, and found that I should use label references instead "fixed" ids, so I tried to change it:

```yaml
  post_id: about
```

This of course didn't work, and embarrassingly took me a while to discover why I was getting a `post_id` of `0`.

While I understand generators are starting point, imho there's an inconsistency: if the generator is **not** creating `id` keys by default, why they are actually being created in `belongs_to` associations? I think it will make more sense that the generated file, in my example `test/fixtures/comments.yml` would be:

```yaml
one:
  commenter: MyString
  body: MyText
  post: 

two:
  commenter: MyString
  body: MyText
  post: 
```

If this makes sense I can submit a PR, though I'm not sure if [generators/test_unit/model/templates/fixtures.yml](https://github.com/rails/rails/blob/master/railties/lib/rails/generators/test_unit/model/templates/fixtures.yml) is the right place to change the logic... and I can't find how is that the generator skips the creation of the `id` key (I only found [a few](https://github.com/rails/rails/blob/master/railties/test/generators/model_generator_test.rb#L278-#L320) fixtures tests, but they didn't helped me).

### PR DESCRIPTION
**Pull request by pablox-cl**
Fixes #18301

Is there anything I could do to improve this, please tell me.

### CODE REVIEW

File: railties/lib/rails/generators/test_unit/model/templates/fixtures.yml [hunk 1]
```diff
@@ -5,6 +5,8 @@
 <% attributes.each do |attribute| -%>
   <%- if attribute.password_digest? -%>
   password_digest: <%%= BCrypt::Password.create('secret') %>
+  <%- elsif attribute.reference? -%>
+  <%= yaml_key_value(attribute.column_name.gsub(/_id$/, ''), attribute.default) %>
```
Comment (rafaelfranca) [inline railties/lib/rails/generators/test_unit/model/templates/fixtures.yml]: We can use `sub` instead of `gsub` here.

**Comment by rafaelfranca** [comment]
Merged as a699f5d3e18e87e802bc460b6e4d2b6fd4921884. Thanks

**Comment by PaBLoX-CL** [comment]
Thanks to you :), hope it's useful! Another question, everything merged into master, goes into the next major release (5.0), right? Since it's a small addition, and one could think it's a bug fix, could it be backported to 4.2.x or 4.3 (if it's going to exist)?

**Comment by simi** [comment]
@PaBLoX-CL 4.3 is not planned and only bugfixes are accepted for 4.2.

**Comment by PaBLoX-CL** [comment]
Okay, no problem :)

**Pull request closed without merging**